### ISSUE TITLE
Philosopher and TPP

### ISSUE THREAD

**Issue by tomthun** [prvst/philosopher#14]
Hi, 

there is a problem with Philosopher and the trans proteomics pipeline (tpp):
I installed tpp some days ago to test it. Afterwards i wanted to compare the results with MSFragger + Philosopher. However the single, separated installation of Philosopher connected with MSFragger didn't work anymore, because it couldn't generate tmp files. The tpp also had problems with generating temporary files. I narrowed the problem down to the mkdir.exe of the tpp installation. To test this i generated folders in the cmd like this `tpp/mkdir.exe -p tmp` which worked. The tpp does something like this `tpp/mkdir.exe -p /tmp` which fails. In conclusion, i think that every Philosopher installation trys to access the mkdir.exe of tpp and generate folders not the way it is supposed to be.  I hope this is interesting for you as well although your software works perfectly fine on its own.

Best,
tom

TL;DR: Philosopher installations try to access the mkdir.exe of tpp and generate folders not the way it is supposed to be.

**Comment by prvst** [comment]
Hey @tomthun ;

Can you provide me with more details on the system you are running the program and the command you tried to run that failed ? If possible please also post here the error message you are getting.

I find this a little odd, we have here a few computers here with TPP and Philosopher and they are working fine (BTW, Philosopher doesn't need to be installed, everything is self-contained in the binary).

By what you are describing it looks to me that TPP is trying to put a directory on your root directory, which is not possible, but I don't see why the program would try to run that on a Linux OS.

**Comment by tomthun** [comment]
@prvst  [Output.txt](https://github.com/prvst/philosopher/files/1723269/Output.txt)

I reinstalled TPP and the problem still persists. The Output of Philosopher is in the Link. (created with MSFragger-GUI) The problem always appears at the same point, after trying to run peptideprophet. This was tested with different mzXML-Files.   
Maybe  @chhh can help out to find which of the default commands of MSFragger GUI for philosopher failed. 

I installed Philosopher externally for MSFragger GUI and not for TPP. (You need the philosopher.exe for MSFragger GUI) I assumed that philosopher is self-contained in the TPP - binary. 

I am running the program on Windows 7 Professional, 64 bit.  I also think that there is a Problem with the directories however i am not sure if it is a problem with the root. 

As i said, after uninstalling TPP the MSFragger-GUI and associated philosopher run perfectly fine again. If you don't have the problem with your computers we can also close this issue as i just need MSFragger GUI + Philosopher to run.

**Comment by chhh** [comment]
@tomthun there's no extra magic in the gui. In the log it says it's gonna execute `D:\ProjectMQMSFRAG\Philosopher\Philosopher20180212.exe peptideprophet --nonparam --expectscore --decoy rev --decoyprobs --masswidth 1000.0 --clevel 2 --database D:\ProjectMQMSFRAG\Philosopher\Taxoplasm_rev.fasta D:\ProjectMQMSFRAG\Data\ConvertData\test\20160824_04_Asp3_TMT10TAILS_npo.pepXML` and that's what it actually is executing. It's a full equivalent of opening a terminal (e.g. **cmd.exe**) changing directory to the output directory that you specified before running and typing that exact command. Can you try that please?

**Comment by chhh** [comment]
@prvst @tomthun I have a feeling that philospher might be starting a process like `peptideprophet ... blah..` and it turns out that `peptideprophet` that's on PATH that is a part of TPP comes up before the one that's pre-packaged? 

@prvst What's the working directory of the process that starts `peptideprophet`? If it's not the temporary directory where the binaries get unpacked, then it surely can be the case. Anyway, I see no other explanation for why TPP installation would interfere.

**Comment by tomthun** [comment]
@chhh I typed the same command in the cmd.exe and the error is still the same.

**Comment by RichardJacoby** [comment]
Hi,

I am having a similar problem, and after email contact with chhh, he asked me to post the details of my problem in this ticket.

When I run MS-Fragger, the open search is completed successfully. But, errors occur when MS-Fragger subsequently proceeds to run PeptideProphet and ProteinProphet.  In both cases, there seems to be a problem in finding the relevant PEPXML and PROTXML files.

For PeptideProphet, the error message reads:

> terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<std::ios_base::failure[abi:cxx11]> >'
>   what():  failed opening file: Das System kann die angegebene Datei nicht finden.: iostream error

 (FYI: “Das System kann die angegebene Datei nicht finden“ is German; it means  “The system cannot find the stated file“.)

And for ProteinProphet, the error message reads:

Error!  Input file not found:

My operating system in Windows 7 Professional with Service Pack 1. I have NOT installed the TPP on this computer.

I am happy to provide more information if it will help.

Thanks,
Richard

**Comment by prvst** [comment]
Hey everyone;

Thanks for the input, I'll investigate the issue and see how to fix this. We have been getting some problems with Windows, mostly because of small particularities from the different versions. For now I can suggest you that if possible, try to run the analysis on a Linux machine.

I'll update the Issue when I get to a solution.

**Comment by prvst** [comment]
@tomthun , @RichardJacoby 
I need you to send me the Philosopher version you are running.

**Comment by chhh** [comment]
@tomthun Thanks Tom, running form the console was just to double check that the problem is somewhere in the philosopher binary.

**Comment by tomthun** [comment]
@prvst I am using this version. https://github.com/prvst/philosopher/releases/tag/20180212

**Comment by RichardJacoby** [comment]
Hi  @prvst,
The first time I got this error, I was using "Philosopher 20180212". I have subsequently downloaded the latest version (Philosopher 20180214), but I am still getting exactly the same error.
Richard

**Comment by prvst** [comment]
@tomthun @RichardJacoby 
Just a quick update here; I tested Philosopher and PeptideProphet on two different Windows machines, a Windows server with TPP already installed and a Windows 10 without TPP. In both cases the program worked correctly without any issues. I'm still going to investigate more whats going on with your analysis, but so far the only issue I noticed on your feedback is that, for some reason PeptideProphet is behaving as if it is on a GNU/Linux environment.

Can you guys run `philosopher workspace --backup` on your workspace folder and send me the zip file ?

**Comment by tomthun** [comment]
@prvst [backup.zip](https://github.com/prvst/philosopher/files/1739371/backup.zip)

Done with Philosopher Version 14.02.2018. The problem still only appears for me if both philosopher and tpp are installed.

**Comment by RichardJacoby** [comment]
Hi @prvst,
This command gives me an error: First it reports "creating backup", but then immediately it reports a WARNING "cannot find the meta data", followed by a FATAL ERROR "cannot find the meta data" . No .ZIP file is created. 
Cheers,
Richard

**Comment by prvst** [comment]
@RichardJacoby 
The workspace is essential to run any analysis using Philosopher, if you don't create one, there is no way you can run the program and any of its functionalities. Please check if you are not executing the command in a different directory or if you deleted the workspace by mistake.

**Comment by prvst** [comment]
@tomthun 
I can see on your analysis meta data that you are using a previous version. Can you try deleting the workspace, downloading the newest version of Philosopher and trying again ?

**Comment by chhh** [comment]
@prvst Where is the "workspace" and how do I delete it? I don't see anything in the output directory.
The GUI inserts the following two commands into execution flow: `philosopher workspace --clean 
` and `philosopher workspace --init` is this not enough?

**Comment by prvst** [comment]
The workspace is the directory where you run the `workspace --init` command. You can remove the folder and start with a new one or just manually remove the `.meta` folder from it. As I mentioned before, I don't understand why the GUI keeps creating and removing the workspace with each command, it only pollutes the output and increases the chance of having issues with the directories used by the program. The `workspace --init` command needs to be the first thing to execute before starting a new analysis and `workspace --clean` should only be executed after backing up the data as it will delete all processed information.

**Comment by chhh** [comment]
@prvst 
- I don't think that answers the question. The GUI runs `workspace --clean`, which I've just tried, and it deleted the `.meta` folder. The GUI also runs `workspace --init`, so I don't see why a workspace left from a previous version of philosopher would interfere, as you replied to @tomthun. Does it mean that we should always manually delete `.meta` when a different version of philosopher is downloaded?
- I also think you should just refer to it as **.meta** folder instead of the mysterious "workspace". Telling people to delete the "workspace" is not very helpful, unlike asking to delete a folder.
- I only added `init/clean` because it was not working without those commands previously, you could have just told me that or opened a ticket. I also don't see what "issues with directories" might be there, if anything, it feels safer to just delete it and start anew, considering there is some sort of interference form previous runs.
- For me this `.meta` directory is always created in the **working directory**, not the output directory. So if you run from command line it's sort of dependent on the path from which you invoke the binary, which is totally unexpected. For GUI there is no such a problem as it always sets the working directory to the location of the binary.

**Comment by tomthun** [comment]
@prvst [backup.zip](https://github.com/prvst/philosopher/files/1747445/backup.zip)

This should be the right one now. I deleted the old `.meta` folder, tried the command which still failed, and saved the `.meta` bin via `--backup`.

**Comment by RichardJacoby** [comment]
Hi @prvst and @chhh,
My priority is to use MS-Fragger to identify interesting PTMs, so for me, it would be good if the MSFragger GUI would instruct Philosopher to automatically write these .meta directories. Indeed, the newest version of MSFragger is doing this for me. 

Unfortunately, I am still receiving error messages related to finding the appropriate files. The error messages read:

> FATA[11:12:40] No pepXML files found, check your files and try again
> and 
> FATA[11:12:40] cannot restore serialized data structures: invalid argument

I am happy to send more information if it will help,
Richard

**Comment by chhh** [comment]
@RichardJacoby That looks to be a different problem. Can you please post the whole output from a run? Do you see pepxml files in the "output directory"?

**Comment by prvst** [comment]
@tomthun 

I don't see any issue with your Philosopher setting, your workspace configuration looks fine. It looks like that the issue truly originates in PeptideProphet, for some reason it only happens in certain windows environments.

I'm planning to do another update on the Prophets implementation Philosopher provides, in order to do that I'll have to work on the Prophets source code and that might take some time. For now I suggest you to try working with the Linux version or on a computer without TPP installed.

I'll post more updates here as soon as I get them.

thanks fro the feedback

**Comment by tomthun** [comment]
@prvst 
I hope you will find the problem soon. 

That's fine with me. Glad to help and jolly debugging.

**Comment by RichardJacoby** [comment]
Hi @chhh,
Here are the outputs from two runs – one with .PEP.XML as output type, the other with .TSV as output type.
[msfragger_output.txt](https://github.com/prvst/philosopher/files/1758485/msfragger_output.txt)

When I select “PEP_XML” as the output type, then MS-Fragger writes a .PEP.XML file in the output directory. But, when I select “TSV” as the output type, then no .PEP.XML file is written.

Cheers, Richard

**Comment by chhh** [comment]
@RichardJacoby Besides any other problems, you're using a buggy version of the GUI (I introduced a bug in v4.5, which was fixed since), you should get a notification that a newer version is available. You can also just go and download it here: https://github.com/chhh/MSFragger-GUI/releases/latest

**Comment by RichardJacoby** [comment]
Hi @chhh and @prvst,
Success! I have downloaded the latest version of MS-Fragger and the latest version of Philosopher. Now everything is working fine. 
Big thanks to both of you for your help with this!
Richard
